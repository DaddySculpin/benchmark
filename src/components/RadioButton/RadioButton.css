.input {
  position: absolute;
  opacity: 0;
  margin: 0;
  width: 0;
  height: 0;
}

.radio {
  display: flex;
  cursor: pointer;
  align-items: center;
  padding: 0.292em 0;
}

.icon {
  margin-right: 0.667em;
}

.disabled {
  cursor: default;
}

.disabled .icon {
  opacity: 0.3;
}
