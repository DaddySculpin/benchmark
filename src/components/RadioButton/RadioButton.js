import React from 'react';
import css from './RadioButton.css';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon/Icon.js';

function RadioButton({
  checked,
  children,
  className,
  disabled,
  id,
  onBlur,
  onClick,
  onChange,
  onFocus,
  value
}) {
  const inputProps = {
    checked,
    disabled,
    id,
    onClick,
    onChange,
    onFocus,
    onBlur,
    value
  };

  const icon = checked ? 'radio-selected' : 'radio-unselected';
  const classes = classnames(className, css.radio, {
    [css.disabled]: disabled
  });

  return (
    <label className={classes}>
      <Icon className={css.icon} iconName={icon} size="medium" />
      <input type="radio" className={css.input} {...inputProps} />
      {children}
    </label>
  );
}

RadioButton.propTypes = {
  checked: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  id: PropTypes.string,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  onClick: PropTypes.func,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string
};

RadioButton.defaultProps = {
  checked: false
};

export default RadioButton;
