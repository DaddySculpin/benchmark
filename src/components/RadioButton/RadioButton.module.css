.input {
  position: absolute;
  opacity: 0;
}

.radio:focus-within {
  outline: 1px dotted;
}

.radio {
  display: flex;
  cursor: pointer;
  align-items: center;
  padding: 0.292em 0;
}

.icon {
  margin-right: 0.667em;
}

.disabled {
  cursor: default;
}

.disabled .icon {
  opacity: 0.3;
}
