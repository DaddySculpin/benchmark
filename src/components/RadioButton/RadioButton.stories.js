import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';

import RadioButton from './RadioButton.js';

storiesOf('RadioButton', module).add('RadioButton', () => {
  const [selected, setSelected] = useState('a');

  return (
    <div role="radiogroup">
      <RadioButton
        value="a"
        name="radio"
        checked={selected === 'a'}
        onChange={setSelected}
      >
        Option A
      </RadioButton>
      <RadioButton
        value="b"
        name="radio"
        checked={selected === 'b'}
        onChange={setSelected}
      >
        Option B
      </RadioButton>
      <RadioButton
        disabled
        value="c"
        name="radio"
        checked={selected === 'c'}
        onChange={setSelected}
      >
        Disabled
      </RadioButton>
    </div>
  );
});
