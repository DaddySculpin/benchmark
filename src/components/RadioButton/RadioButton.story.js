import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';

import RadioButton from './RadioButton';

storiesOf('Classic', module).add('Elements: Radio Button', () => {
  const [selected, setSelected] = useState();

  function handleChange(e) {
    setSelected(e.target.value);
  }

  return (
    <div role="radiogroup">
      <RadioButton
        value="a"
        name="radio"
        checked={selected === 'a'}
        onChange={handleChange}
      >
        Option A
      </RadioButton>
      <RadioButton
        value="b"
        name="radio"
        checked={selected === 'b'}
        onChange={handleChange}
      >
        Option B
      </RadioButton>
      <RadioButton
        disabled
        value="c"
        name="radio"
        checked={selected === 'c'}
        onChange={handleChange}
      >
        Disabled
      </RadioButton>
    </div>
  );
});
