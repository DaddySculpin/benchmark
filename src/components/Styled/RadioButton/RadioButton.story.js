import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';

import { Global, ThemeProvider } from '../../../index.js';
import RadioButton from './RadioButton.js';

storiesOf('RadioButton', module).add('RadioButton', () => {
  const [selected, setSelected] = useState('a');

  return (
    <Global>
      <ThemeProvider>
        <div role="radiogroup">
          <RadioButton
            value="a"
            name="radio"
            checked={selected === 'a'}
            onChange={setSelected}
          >
            Option A
          </RadioButton>
          <RadioButton
            value="b"
            name="radio"
            checked={selected === 'b'}
            onChange={setSelected}
          >
            Option B
          </RadioButton>
          <RadioButton
            disabled
            value="c"
            name="radio"
            checked={selected === 'c'}
            onChange={setSelected}
          >
            Disabled
          </RadioButton>
        </div>
      </ThemeProvider>
    </Global>
  );
});
