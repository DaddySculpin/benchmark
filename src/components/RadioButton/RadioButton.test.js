import React from 'react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { render } from '@testing-library/react';
import RadioButton from './RadioButton';

expect.extend(toHaveNoViolations);

function handleChange() {
  console.log('change');
}

describe('RadioButton', () => {
  it('renders without crashing', () => {
    render(<RadioButton onChange={handleChange} />);
  });

  it('should not have basic accessibility issues', async () => {
    const { container } = render(<RadioButton />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  xit('should have a label element wrapping the input', () => {
    throw new Error();
  });

  it('should show a radio-selected icon when checked', () => {
    const { container } = render(
      <RadioButton checked onChange={handleChange} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should show a radio-selected icon when checked', () => {
    const { container } = render(
      <RadioButton checked onChange={handleChange} />
    );
    const selected = container.querySelector('.radio-selected');
    expect(selected).toBeTruthy();
  });

  it('should show a radio-unselected icon when not checked', () => {
    const { container } = render(<RadioButton onChange={handleChange} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  xit('should change the icon opacity when disabled', () => {
    throw new Error();
  });

  xit('should not change the text opacity when disabled', () => {
    throw new Error();
  });

  xit('should not change the text opacity when disabled', () => {
    throw new Error();
  });

  xit('should show a focus ring when focused', () => {
    throw new Error();
  });

  xit('should not show a focus ring when unfocused', () => {
    throw new Error();
  });

  xit('should invoke the change handler', () => {
    throw new Error();
  });

  xit('should invoke the focus handler', () => {
    throw new Error();
  });

  xit('should invoke the blur handler', () => {
    throw new Error();
  });
});
