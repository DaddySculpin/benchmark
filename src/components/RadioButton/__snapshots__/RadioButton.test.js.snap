// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should show a radio-selected icon when checked 1`] = `
<label
  class="radio"
>
  <span
    class="icon icon radio-selected medium"
    data-icon=""
  />
  <input
    checked=""
    class="input"
    type="radio"
    value=""
  />
</label>
`;

exports[`RadioButton should show a radio-unselected icon when not checked 1`] = `
<label
  class="radio"
>
  <span
    class="icon icon radio-unselected medium"
    data-icon=""
  />
  <input
    class="input"
    type="radio"
    value=""
  />
</label>
`;
